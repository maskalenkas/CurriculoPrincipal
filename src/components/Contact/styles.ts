import styled, { css, DefaultTheme } from 'styled-components'
import media from 'styled-media-query'

import { StyledIconBase } from '@styled-icons/styled-icon'

import { ContactCard } from '@styled-icons/fluentui-system-filled'
import { Whatsapp } from '@styled-icons/boxicons-logos'

import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    /* Selecionando o icone de agenda */
    ${HeadingStyles.Wrapper} {
      text-align: center;
      margin-top: 1rem;
    }
  `}
`

// Centralizando tudo
export const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`

// Molde dos icones
const BaseIconModifier = {
  default: (
    theme: DefaultTheme,
    colorIcon: 'green' | '#0487cc',
    sizeIcon?: string,
  ) => css`
    color: ${colorIcon};
    width: ${sizeIcon};
  `,
}

// Icones
export const IconContactCard = styled(ContactCard)`
  ${({ theme }) => css`
    ${BaseIconModifier.default(theme, '#0487cc', '15rem')}

    ${media.lessThan('medium')`
      ${BaseIconModifier.default(theme, '#0487cc', '10rem')}
    `}
  `}
`

export const IconWhatsapp = styled(Whatsapp)`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.xsmall};
    ${BaseIconModifier.default(theme, 'green', '10rem')}
    ${media.lessThan('medium')`
      ${BaseIconModifier.default(theme, 'green', '7rem')}
    `};
  `}
`
