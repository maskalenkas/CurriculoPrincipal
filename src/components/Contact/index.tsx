import * as S from './styles'

import Heading from 'components/Heading'
import Icons from 'components/Icons'

import { ContactCard } from '@styled-icons/fluentui-system-filled'
import { Whatsapp } from '@styled-icons/boxicons-logos'

const Contact = () => (
  <S.Wrapper>
    <Icons
      color="primary"
      size="15rem"
      sizeMedia="12rem"
      isScale
      title="icon Contact"
    >
      <ContactCard />
    </Icons>

    <S.Content>
      <Heading
        asTextType="title"
        color="secondary"
        weight="bold"
        size="xxlarge"
      >
        Vamos criar algo incrivel?
      </Heading>
      <Heading asTextType="subtitle" color="secondary">
        A melhor forma de contato é através do meu Whatsapp
      </Heading>

      {/* Gambiarra para não tipar icons */}
      <div
        onClick={() =>
          (window.location.href =
            'https://wa.me/+55011976660220?text=Tudo%20Bem?')
        }
      >
        <Icons
          color="green"
          size="10rem"
          sizeMedia="7rem"
          marginTop="1.5rem"
          isScale
          title="icon Whatsapp"
        >
          <Whatsapp />
        </Icons>
      </div>
    </S.Content>
  </S.Wrapper>
)

export default Contact
