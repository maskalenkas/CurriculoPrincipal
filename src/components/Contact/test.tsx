import { render, screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Contact from '.'

describe('<Contact />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Contact />)
  })
})
