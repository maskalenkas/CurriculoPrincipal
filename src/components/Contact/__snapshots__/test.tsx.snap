// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Contact
  </h1>
</main>
`;
